src/components/Stepper/Step/Step.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { IconName } from '@talend/icons';
+import { Icon } from '../../Icon/Icon';
+
+import * as S from './Step.style';
+
+export type StepProps = {
+	/** The title of the step */
+	title: string;
+	/** The index displayed in front of the step */
+	index: string;
+	/** The optionnal class name */
+	className: string;
+	/** The icon element to display */
+	icon: IconName;
+};
+
+/**
+ Steps are the main elements for the stepper.
+ * */
+
+const Step: React.FC<StepProps> = ({ icon, index, title, className = '', ...rest }: StepProps) => {
+	return (
+		<S.Step role="status" aria-live="polite" {...rest} className={`step ${className || ''}`}>
+			<span className="step__index">{index}</span>
```

Comment:
The step index doesn't need to appear in HTML, it's some CSS decoration.


```suggestion
```

It's more a pseudo-element
```css
.step[data-index]::before { content: attr(data-index) ". "; }
```